### ISSUE TITLE
Spyder 6 stuck when computing `xHeight` of monospace font

### ISSUE THREAD

**Issue by xaverm** [spyder-ide/spyder#22661]
## Problem Description
Spyder versions 6.0.0 / 6.0.1 hang within the startup, not reaching the GUI but only producing 100% CPU load.

### What steps reproduce the problem?
`conda install -c conda-forge spyder=6.0.0  or 6.0.1`
Invoke `spyder` on the CLI.
Instead of the GUI starting up, the command does not complete, but is simply stuck, with no output resulting.
In the `top`-list, one can see that a spyder processes has started and produces 100% CPU load.
Only `CTRL-C` on the CLI (or killing the process) will stop this.
Upon stopping spyder, one can see in the CLI that the most recent call consistently is
```
 "..../envs/py/lib/python3.9/site-packages/spyder/utils/qthelpers.py", line 842, in set_monospace_interface_font
     QFontMetrics(plain_font).xHeight() != x_height
```
I'm happy to provide more info if you tell me how in this situation.
Spyder version 5.5.5 starts with no problem.

* Spyder version:  6.0.0 or 6.0.1
* Python version:  3.9.20
* Qt version: qt-main 5.15.8
* PyQt version: 5.15.9 
* Operating System name/version: linux

(I can't check spyders' 6.0.0 and 6.0.1 dependencies, as they don't start ...)

**Comment by xaverm** [comment]
I am aware that my initial issue description may have been somewhat vague. Therefore  I have changed the title and added some (hopefully) clarifying remarks.
The issue I am facing is that versions 6.0.0 and 6.0.1 do not run within a freshly set up environment with an updated base environment.

What I do to get this is:
1) I am not using the base env, but anyway, just to make sure, I do `conda activate base` and then `conda update --all` and then `conda deactivate`
2) Next I do `conda create -n sptes python=3.9`, then `conda activate sptes`, then `conda install -c conda-forge spyder=6.0.1`
3) Finally, without any further ado, I invoke `spyder` on the CLI of this freshly generated `sptes` env.
4) After that `spyder` does not start a GUI but just runs with 100% CPU load in the `top`-list. This is the same behavior I reported previously from within my actual envs.

I don't know if it helps, but my OS is an openSUSE Leap linux.

**Comment by xaverm** [comment]
B.t.w.:
This issue is accompanied by weird additional side effects: After returning to version 5.5.5, which is working, I find that versions 6.0.0 and 6.0.1, for each attempted, yet malfunctioning invocation, do modify spyder's user preferences in `spyder.ini` without telling the user so. I.e. in my case they change `editor/run selection = Shift+Return` to `editor/run selection = F9` although they do not start the GUI. In my case this produces a conflict with another preference once I'm back to the functioning older version 5.5.5. Are such hidden changes to `spyder.ini` really necessary?

**Comment by ccordoba12** [comment]
Hey @xaverm, thanks for reporting. Please run `spyder --reset` and try again (another person reported that that fixed this problem).

If not, I'm afraid you're using a monospace font for which Spyder is unable to get it's height as expected, which is the reason why it gets stuck in an infinite loop.

In any case, we'll try to fix this bug in 6.0.3 or 6.0.4, both to be released before the end of the year.

You also said:

> This issue is accompanied by weird additional side effects: After returning to version 5.5.5, which is working, I find that versions 6.0.0 and 6.0.1, for each attempted, yet malfunctioning invocation, do modify spyder's user preferences in spyder.ini without telling the user so.

That's a side effect of going back to a previous version. Since some options changed between Spyder 5 and 6, they are reset to their default values. I don't think we're going to address that because things have being working like that for close to 10 years.

**Comment by xaverm** [comment]
Hi @ccordoba12: Yes, I was using a monospace font in 5.5.5 at `Preferences->Appearance->Fonts->Plain text` for what is called `Preferences->Appearance->Fonts->Monospace` in 6.0.1 today. For now, I've changed that font into something else from within a running 5.5.5 and that solved the issue for me in 6.0.1 (no `spyder --reset` needed.)
(B.t.w.: The width of the drop down list for fonts in the `Preferences` of 6.0.1 is rather narrow, i.e., some font names get truncated as compared to 5.x.y.)

> You also said:
> > This issue is accompanied by weird additional side effects
>
> things have being working like that for close to 10 years and no one has complained so far.

So I've been experiencing this also for close to 10 years - and never realized it/complained - I'd never complain about spyder anyway.
Keep it up and running!

**Comment by ccordoba12** [comment]
> For now, I've changed that font into something else from within a running 5.5.5 and that solved the issue for me in 6.0.1 (no spyder --reset needed.)

Great! Glad you were able to do that without resetting your current options.

> (B.t.w.: The width of the drop down list for fonts in the Preferences of 6.0.1 is rather narrow, i.e., some font names get truncated as compared to 5.x.y.)

Please open a new issue about it, attaching a screenshot showing the problem to it, so we don't forget to take a look at it.

> So I've been experiencing this also for close to 10 years - and never realized it/complained - I'd never complain about spyder anyway.

We reset default option values rarely, so that's probably why you didn't notice that behavior before.
